chia/plotters/bladebit.py

```diff
@@ -275,45 +272,47 @@ def plot_bladebit(args, chia_root_path, root_path):
         call_args.append("-v")
     if args.nonuma:
         call_args.append("-m")
-    if args.memo is not None and args.memo != b"":
+    if "memo" in args and args.memo is not None and args.memo != b"":
```

Comment:
What is making the parameters maybe not present now?

Reply:
`args` here is `arpparse`'s `Namespace` object and it raises error if specified property does not exist.
`memo` is an option of bladebit2 and bladebit1 does not expect the option.

https://github.com/Chia-Network/chia-blockchain/blob/cmj.bb2-ramplot/chia/plotters/plotters.py#L91-L131
Check these lines for your convinience.
`--memo` is an option for bladebit2 but not for bladebit1.

Reply:
I would expect we won't be developing bladebit1 anymore so having the code support it just stay and remain unchanged seems ok.  New bladebit2 code can handle just bladebit2 without having to worry about backwards compatibility with 1.  Also, checking this way cascades from version selection to arg definition to arg usage rather than just using the single layer of version selection as the condition everywhere.  But, I admittedly have my whole own other way that I've handled plotters in plotman so... `:]`

I won't block merging on this topic, feel free to close this conversation if you would rather keep this all as is.

Reply:
Thanks for your comment Kyle.
I was thinking whether I should create another PR to drop bladebit1 support but I decide to manage that in this PR.